src/identify.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Wrapper for a user properties JSON object that supports operations
+ */
+
+var AMP_OP_ADD = '$add';
+var AMP_OP_SET = '$set';
+var AMP_OP_SET_ONCE = '$setOnce';
+var AMP_OP_UNSET = '$unset';
+
+
+var Identify = function() {
+  this.userPropertiesOperations = {};
+  this.properties = []; // keep track of keys that have been added
+};
+
+var isNumeric = function(n) {
+  return !isNaN(parseFloat(n)) && isFinite(n);
+};
+
+Identify.prototype.add = function(property, value) {
+  if (isNumeric(value) || typeof(value) === 'string' || value instanceof String) {
```

Comment:
They could do something like $add "3". On server side, we do string to number validation: https://github.com/amplitude/amplitude/blob/master/amp/model/user.py#L146, although I can do similar things on the SDK as well